app/services/exception_safe_response_parser.rb

```diff
@@ -27,7 +27,7 @@ def message_for(response, ssl_error)
     if response.is_a? Exception
       exception_message(response)
     else
-      if ssl_error
+      if ssl_error && response.data.is_a?(String)
```

Comment:
It feels a bit weird to do the Type check here when there's another one in `#normal_message`.  Would you consider pushing this further into the parser, by doing the conditional concatenation in `#code_and_data` which is only called when `data` is not a Hash and is, therefore, most likely a String.

Reply:
@EdwardAndress refactored.